easybuild/easyblocks/m/matlab.py

```diff
@@ -188,13 +191,20 @@ def install_step(self):
             # check installer output for known signs of trouble
             patterns = [
                 "Error: You have entered an invalid File Installation Key",
+                "Not a valid key",
+                "Exiting with status -2",
```

Comment:
should we make this more general so other non-zero exit codes are also catched?

Reply:
well i wasn't sure what codes where actual errors or not, like -1 maybe is a only a warning or so?

Reply:
looks like -1 is returned if already installed
https://www.mathworks.com/matlabcentral/answers/1953454-updater-shows-new-version-of-matlab-2023a-however-clicking-the-updater-button-doesn-t-do-anything

but it also prints "End - Unsuccessful", so not really needed i guess

Reply:
here it returns -1 when there write permission issues
https://ww2.mathworks.cn/matlabcentral/answers/334889-can-t-install-any-toolboxes-because-can-t-write-to-usr-local-matlab-r2017